proxy/fastcgi/proxy.go

```diff
@@ -0,0 +1,210 @@
+package fastcgi
+
+import (
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/fabiolb/fabio/config"
+	"github.com/fabiolb/fabio/route"
+)
+
+var (
+	headerNameReplacer = strings.NewReplacer(" ", "_", "-", "_")
+)
+
+type Proxy struct {
+	upstream *route.Target
+
+	// ProxyConfig is the proxy configuration as provided during startup.
+	ProxyConfig config.Proxy
+
+	// FastCGIConfig is the FastCGI configuration as provided during startup.
+	FastCGIConfig config.FastCGI
+}
+
+func NewFCGIProxy(target *route.Target, proxyConfig *config.Proxy, fcgiConfig *config.FastCGI) *Proxy {
```

Comment:
You only need the host from the target and I usually just pass in the entire config object.

```go
func NewFCGIProxy(cfg *config.Config, upstream string) *Proxy { ... }
```